### ISSUE TITLE
REAR not creating bootable USB on Dell XPS15-9500 dual boot Ubuntu 20.04 and Windows 10

### ISSUE THREAD

**Issue by hi-flyer** [rear/rear#2556]
#### Relax-and-Recover (ReaR) Issue Template

Fill in the following items before submitting a new issue
(quick response is not guaranteed with free support):

* ReaR version ("/usr/sbin/rear -V"):
Relax-and-Recover 2.5 / Git

* OS version ("cat /etc/os-release" or "lsb_release -a" or "cat /etc/rear/os.conf"):
```
NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.1 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

* ReaR configuration files ("cat /etc/rear/site.conf" and/or "cat /etc/rear/local.conf"):
```
# Default is to create Relax-and-Recover rescue media as ISO image
# set OUTPUT to change that
# set BACKUP to activate an automated (backup and) restore of your data
# Possible configuration values can be found in /usr/share/rear/conf/default.conf
#
# This file (local.conf) is intended for manual configuration. For configuration
# through packages and other automated means we recommend creating a new
# file named site.conf next to this file and to leave the local.conf as it is. 
# Our packages will never ship with a site.conf.
### write the rescue initramfl to USB and update the USB boot loader
OUTPUT=USB

### Create a backup using the internal NETFS method, using tar
BACKUP=BAREOS
BAREOS-CLIENT=bareos-fd

#Write bothe rescue image and backup to the device labeled REAR-000
BACKUP_URL=usb:///dev/disk/by-label/REAR-000
EOF
```
* Hardware (PC or PowerNV BareMetal or ARM) or virtual machine (KVM guest or PoverVM LPAR):
Dell XPS15-9550 PC

* System architecture (x86 compatible or PPC64/PPC64LE or what exact ARM device):
x86 compatible

* Firmware (BIOS or UEFI or Open Firmware) and bootloader (GRUB or ELILO or Petitboot):
BIOS (dell 1.14.0) GRUB2 (2.04-1ubuntu26.7)
* Storage (local disk or SSD) and/or SAN (FC or iSCSI or FCoE) and/or multipath (DM or NVMe):
SSD - THNSN5512GPU7 NVMe TOSHIBA 512GB
* Storage layout ("lsblk -ipo NAME,KNAME,PKNAME,TRAN,TYPE,FSTYPE,SIZE,MOUNTPOINT" or "lsblk" as makeshift):
```
NAME        MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
loop0         7:0    0    55M  1 loop /snap/core18/1880
loop1         7:1    0   219M  1 loop /snap/gnome-3-34-1804/66
loop2         7:2    0  55.4M  1 loop /snap/core18/1944
loop3         7:3    0 217.9M  1 loop /snap/gnome-3-34-1804/60
loop4         7:4    0    51M  1 loop /snap/snap-store/518
loop5         7:5    0  62.1M  1 loop /snap/gtk-common-themes/1506
loop6         7:6    0  49.8M  1 loop /snap/snap-store/467
loop7         7:7    0  31.1M  1 loop /snap/snapd/10492
loop8         7:8    0  31.1M  1 loop /snap/snapd/10707
loop9         7:9    0  64.8M  1 loop /snap/gtk-common-themes/1514
sda           8:0    1  29.2G  0 disk 
├─sda1        8:1    1   400M  0 part 
└─sda2        8:2    1  28.9G  0 part /media/beckstein/REAR-0002
nvme0n1     259:0    0   477G  0 disk 
├─nvme0n1p1 259:1    0   200M  0 part /boot/efi
├─nvme0n1p2 259:2    0    16M  0 part 
├─nvme0n1p3 259:3    0 197.7G  0 part /media/windows
├─nvme0n1p4 259:4    0     1M  0 part 
├─nvme0n1p5 259:5    0  15.9G  0 part 
├─nvme0n1p6 259:6    0    11G  0 part 
├─nvme0n1p7 259:7    0   513M  0 part 
├─nvme0n1p8 259:8    0   513M  0 part 
└─nvme0n1p9 259:9    0 250.8G  0 part /
```
* Description of the issue (ideally so that others can reproduce it):
* REAR does create the bootable usb but appears to have most of all required info but on the USB for same laptop, the usb does not show up in the bootable devices listed. after multiple attempts and even creating ISO's, balenaEtcher stated the the ISO does not have the partition information. So while I am trying to write directly to the USB, what is missing that it is not a bootable USB when used on the same laptop.
*  I was required by REAR to format the USB with sudo rear format -- --efi /dev/sda
* I was required by rear to make sure that bareos-dir was running
* I was required to create a XPS15 named fileset in bareos configuration (bareos was installed as the default setup with no modification)
* This is a dual boot Windows 10 and Ubuntu 20.04 system
* have tried on both scandisk (32 GB) and pny (64GB) USBs

* Workaround, if any:
* none found

* Attachments, as applicable ("rear -D mkrescue/mkbackup/recover" debug log files):
[rear-BEE-XPS15.log](https://github.com/rear/rear/files/5857060/rear-BEE-XPS15.log)

```
rear -v dump
Relax-and-Recover 2.5 / Git
Running rear dump (PID 7097)
Using log file: /var/log/rear/rear-BEE-XPS15.log.lockless
# Begin dumping out configuration and system information:
# This is a 'Linux-x86_64' system, compatible with 'Linux-i386'.
# Configuration tree:
  # Linux-i386.conf : OK
  # GNU/Linux.conf : OK
  # Debian.conf : missing/empty
  # Debian/i386.conf : missing/empty
  # Debian/20.04.conf : missing/empty
  # Debian/20.04/i386.conf : missing/empty
  # Ubuntu.conf : OK
  # Ubuntu/i386.conf : missing/empty
  # Ubuntu/20.04.conf : missing/empty
  # Ubuntu/20.04/i386.conf : missing/empty
  # site.conf : missing/empty
  # local.conf : OK
# System definition:
  ARCH="Linux-i386"
  OS="GNU/Linux"
  OS_MASTER_VENDOR="Debian"
  OS_MASTER_VERSION="20.04"
  OS_MASTER_VENDOR_ARCH="Debian/i386"
  OS_MASTER_VENDOR_VERSION="Debian/20.04"
  OS_MASTER_VENDOR_VERSION_ARCH="Debian/20.04/i386"
  OS_VENDOR="Ubuntu"
  OS_VERSION="20.04"
  OS_VENDOR_ARCH="Ubuntu/i386"
  OS_VENDOR_VERSION="Ubuntu/20.04"
  OS_VENDOR_VERSION_ARCH="Ubuntu/20.04/i386"
# Backup with BAREOS:
  BACKUP_DUPLICITY_NAME="rear-backup"
  BACKUP_INTEGRITY_CHECK=""
  BACKUP_MOUNTCMD=""
  BACKUP_ONLY_EXCLUDE="no"
  BACKUP_ONLY_INCLUDE="no"
  BACKUP_OPTIONS=""
  BACKUP_RESTORE_MOVE_AWAY_DIRECTORY="/var/lib/rear/moved_away_after_backup_restore/"
  BACKUP_RESTORE_MOVE_AWAY_FILES=("/boot/grub/grubenv" "/boot/grub2/grubenv")
  BACKUP_RSYNC_OPTIONS=("--sparse" "--archive" "--hard-links" "--numeric-ids" "--stats")
  BACKUP_SELINUX_DISABLE="1"
  BACKUP_TYPE=""
  BACKUP_UMOUNTCMD=""
  BACKUP_URL="usb:///dev/disk/by-label/REAR-000"
# Output to USB:
  USB_DEVICE=""
  USB_DEVICE_FILESYSTEM="ext3"
  USB_DEVICE_FILESYSTEM_LABEL="REAR-000"
  USB_DEVICE_FILESYSTEM_PARAMS=""
  USB_DEVICE_FILESYSTEM_PERCENTAGE="100"
  USB_DEVICE_PARTED_LABEL="msdos"
  USB_PARTITION_ALIGN_BLOCK_SIZE="8"
  USB_RETAIN_BACKUP_NR="2"
  USB_SUFFIX=""
  USB_UEFI_PART_SIZE="400"
  OUTPUT_EFISTUB_SYSTEMD_BOOTLOADER="/usr/lib/systemd/boot/efi/systemd-bootx64.efi"
  OUTPUT_MOUNTCMD=""
  OUTPUT_OPTIONS=""
  OUTPUT_PREFIX="BEE-XPS15"
  OUTPUT_PREFIX_PXE=""
  OUTPUT_UMOUNTCMD=""
  OUTPUT_URL=""
# Validation status:
  # /usr/share/rear/lib/validated/Ubuntu/20.04/i386.txt : missing/empty
  # Your system is not yet validated. Please carefully check all functions
  # and create a validation record with 'rear validate'. This will help others
  # to know about the validation status of Relax-and-Recover on this system.
# End of dump configuration and system information.
Saving /var/log/rear/rear-BEE-XPS15.log.lockless as /var/log/rear/rear-BEE-XPS15.log
Exiting rear dump (PID 7097) and its descendant processes ...
Running exit tasks

/etc/rear/local
# Default is to create Relax-and-Recover rescue media as ISO image
# set OUTPUT to change that
# set BACKUP to activate an automated (backup and) restore of your data
# Possible configuration values can be found in /usr/share/rear/conf/default.conf
#
# This file (local.conf) is intended for manual configuration. For configuration
# through packages and other automated means we recommend creating a new
# file named site.conf next to this file and to leave the local.conf as it is. 
# Our packages will never ship with a site.conf.
### write the rescue initramfl to USB and update the USB boot loader
OUTPUT=USB

### Create a backup using the internal NETFS method, using tar
BACKUP=BAREOS
BAREOS-CLIENT=bareos-fd

#Write bothe rescue image and backup to the device labeled REAR-000
BACKUP_URL=usb:///dev/disk/by-label/REAR-000

EOF

rear -v format -- --efi /dev/sda
Relax-and-Recover 2.5 / Git
Running rear format (PID 10843)
Using log file: /var/log/rear/rear-BEE-XPS15.log
USB device /dev/sda is not formatted with ext2/3/4 or btrfs filesystem
Type exactly 'Yes' to format /dev/sda with ext3 filesystem
(default 'No' timeout 300 seconds)
Yes
Repartitioning '/dev/sda'
The --efi toggle was used with format - making an EFI bootable device '/dev/sda'
Creating GUID partition table (GPT) on '/dev/sda'
Creating EFI system partition with size 400 MiB aligned at 8 MiB on '/dev/sda'
Creating ReaR data partition up to 100% of '/dev/sda'
Setting 'boot' flag on /dev/sda
Creating vfat filesystem on EFI system partition on '/dev/sda1'
Creating ext3 filesystem with label 'REAR-000' on '/dev/sda2'
Adjusting filesystem parameters on '/dev/sda2'
Exiting rear format (PID 10843) and its descendant processes ...
Running exit tasks

rear -v mkrescue
Relax-and-Recover 2.5 / Git
Running rear mkrescue (PID 13908)
Using log file: /var/log/rear/rear-BEE-XPS15.log
Using UEFI Boot Loader for Linux (USING_UEFI_BOOTLOADER=1)
Using autodetected kernel '/boot/vmlinuz-5.8.0-38-generic' as kernel in the recovery system
Creating disk layout
Docker is running, skipping filesystems mounted below Docker Root Dir /var/lib/docker
Using guessed bootloader 'EFI' (found in first bytes on /dev/nvme0n1)
Verifying that the entries in /var/lib/rear/layout/disklayout.conf are correct ...
Creating root filesystem layout
Skipping 'docker0': not bound to any physical interface.
Cannot include default keyboard mapping (no KEYMAPS_DEFAULT_DIRECTORY specified)
Cannot include keyboard mappings (neither KEYMAPS_DEFAULT_DIRECTORY nor KEYMAPS_DIRECTORIES specified)
To log into the recovery system via ssh set up /root/.ssh/authorized_keys or specify SSH_ROOT_PASSWORD
Trying to find what to use as UEFI bootloader...
Trying to find a 'well known file' to be used as UEFI bootloader...
Using '/boot/efi/EFI/Boot_test/grubx64.efi' as UEFI bootloader file
Copying logfile /var/log/rear/rear-BEE-XPS15.log into initramfs as '/tmp/rear-BEE-XPS15-partial-2021-01-22T10:20:44-05:00.log'
Copying files and directories
Copying binaries and libraries
Copying all kernel modules in /lib/modules/5.8.0-38-generic (MODULES contains 'all_modules')
Copying all files in /lib*/firmware/
Symlink '/lib/modules/5.8.0-38-generic/build' -> '/usr/src/linux-headers-5.8.0-38-generic' refers to a non-existing directory on the recovery system.
It will not be copied by default. You can include '/usr/src/linux-headers-5.8.0-38-generic' via the 'COPY_AS_IS' configuration variable.
Symlink '/usr/share/misc/magic' -> '/usr/share/file/magic' refers to a non-existing directory on the recovery system.
It will not be copied by default. You can include '/usr/share/file/magic' via the 'COPY_AS_IS' configuration variable.
Testing that the recovery system in /tmp/rear.mXT6DtwftcxLByt/rootfs contains a usable system
Creating recovery/rescue system initramfs/initrd initrd.cgz with gzip default compression
Created initrd.cgz with gzip default compression (371792610 bytes) in 42 seconds
Saved /var/log/rear/rear-BEE-XPS15.log as rear/BEE-XPS15/20210122.1020/rear-BEE-XPS15.log
Writing MBR of type msdos to /dev/sda
Copying resulting files to usb location
Saving /var/log/rear/rear-BEE-XPS15.log as rear-BEE-XPS15.log to usb location
Copying result files '/tmp/rear.mXT6DtwftcxLByt/tmp/VERSION /tmp/rear.mXT6DtwftcxLByt/tmp/README /tmp/rear.mXT6DtwftcxLByt/tmp/rear-BEE-XPS15.log' to /tmp/rear.mXT6DtwftcxLByt/outputfs/BEE-XPS15 at usb location
Exiting rear mkrescue (PID 13908) and its descendant processes ...
Running exit tasks

```

**Comment by hi-flyer** [comment]
I found some issues but still not showing up on XPS15 9500 boot menu.
First, I uninstalled the rear (was installed by apt install rear)
I then followed website instruction for git clone  and followon instructions
Still had to format usb with -- --efi
the software created a usb on /dev/sdg
I can boot this usb on a Dell XPS15 9500 computer (it shows up as an efi drive)
But the Dell XPS15 9550 (older computer) using latest bios 1.14.0 still the USB does not show up on the boot screen. 
BIOS is set for not secure booting with ACHI. (although I have tried secure boot as well)
the 9550 sees live upbuntu usb just fine. so very confused on this problem.

Any suggestion on how to get the USB to show up on my XPS15-9550?

**Comment by hi-flyer** [comment]
Update
I can now discover the USB on the xps15-9550. I had to delete all predefined options for booting in the bios menu and it then discovered the scandisk usb. I then restored the options I had deleted. the scandisk was still showing on rebooting.
No when booting to the scandisk with the rear info, it starts by loading ramdisk but the second line says cannot allocate initrd.
Any Ideas on what I should do to fix that problem.
TIA
